### ISSUE TITLE
search_lightcurvefile returning files for multiple Kepler IDs

### ISSUE THREAD

**Issue by ceb8** [KeplerGO/lightkurve#718]
#### Problem description
Using `search_lightcurvefile` to look for data from one Kepler ID returns files for a different Kepler ID in addition to the requested one.

#### Example

```python
import lightkurve
lc_filelist = lightkurve.search_lightcurvefile(f"KIC 10058374", cadence='long', quarter=[10]).download_all()
```

#### Expected behavior
I expect:
```python
print(lc_filelist)

LightCurveFileCollection of 2 objects:
	KIC 10058374 (1 KeplerLightCurveFiles) Quarters: 10
```

Instead I get:
```python
print(lc_filelist)

LightCurveFileCollection of 2 objects:
	KIC 10058374 (1 KeplerLightCurveFiles) Quarters: 10
	KIC 10058376 (1 KeplerLightCurveFiles) Quarters: 10
```

#### Environment

-  platform (e.g. Linux, OSX, Windows): OSX High Sierra
-  lightkurve version (e.g. 1.0b6): 1.9.1
-  installation method (e.g. pip, conda, source): pip

**Comment by barentsen** [comment]
@ceb8 Thanks for opening an issue!

KIC 10058374 and KIC 10058376 are ~2 arcseconds apart.  When I look them up in the MAST Portal's graphical display, I find that the footprints of the Kepler observations show a great deal of overlap:

![Screenshot from 2020-03-27 11-03-10](https://user-images.githubusercontent.com/817669/77786201-9fc26800-701a-11ea-851d-a75c74edd2b8.png)

The default behavior of the `search` functions is to search by coordinates (i.e. by passing `objectname` into `Observations.query_criteria`).  My understanding is that the search results returned by the Portal API are based on whether or not a coordinate falls in or near a footprints. KIC 10058374 and KIC 10058376 are both fully contained within each other's footprint, so the portal returns both if you search either coordinate/object.

Of course we could be smarter and pass a specific `target_name` to `Observations.query_criteria`.  The difference between searching via `target_name` or `object_name` gets tricky and confusing quickly, so my thinking has been that it's best to always search by coordinates (i.e. `object_name`), so that a user is forced to pay attention if their target is covered by multiple overlapping pixel files.  Is that the right philosophy?!

For now, a quick fix for your issue would be to filter the search result by target name before the download, for example:
```python
import lightkurve as lk
search = lk.search_lightcurvefile("KIC 10058374", cadence='long', quarter=[10])
search[search.target_name == 'kplr10058374'].download_all()
```

Does that help?

**Comment by ceb8** [comment]
@barentsen For this particular situation I solved it by checking all my detrended light curves for strictly increasing time values, and then redoing the ones that didn't have that with a check for Kepler ID before detrending.

I see why this behavior happens, however the docstring claims "This function fetches a data table that lists the Light Curve Files that fall within a region of sky centered around the position of `target` and within a cone of a given `radius`. If no value is provided for `radius`, only a single target will be returned." So I had assumed that the function itself did whatever checking it needed to do to make sure that I only got results for the target I specified.  

From a user perspective if I am providing a Kepler ID and list of quarters I assume I will be getting data only for that Kepler ID and list of quarters, so I think that at least when `radius=None` it would be good for LightKurve to do the filtering for the user.

In general, I had assumed this function was using `query_criteria` and `target_name`, however I hadn't checked the docstring until I encountered this issue, at which point I realized that it was essentially doing a cone search. It might be interesting to experiment with objectname vs target_name for performance, because from the database perspective the searches are quite different, but I also understand that using objectname might just be more robust.  And also, of course you would then lose the flexibility of using the radius to search nearby targets too.

**Comment by barentsen** [comment]
@ceb8 I hear you!  I agree that if someone specifies a KIC or a TIC idea we should probably be smarter and search by `target_name`.  I hope it won't be confusing tho, because the behavior will be different when e.g. searching by the star's colloquial name.

**Comment by ceb8** [comment]
@barentsen  Ooooh yeah. That makes sense. Hmmmmm. Well if the behavior doesn't change, the docstring should be updated to indicate that because of the density of the field a singular object return cannot be guaranteed.

**Comment by barentsen** [comment]
For simplicity, maybe the best solution is to *always* do a cone search via `object_name` (and explain that clearly in the docs), and have an optional argument to filter by `target_name`, e.g.
```python
search_lightcurvefile("KIC 10058374", strict_target_name=True) 
```

You did make a good point that `target_name` may or may not be much faster to search on than `object_name`.  It'd be worth investigating if there's a significant difference!

**Comment by astrobel** [comment]
Just dropping in to say I found another example of this issue with some open cluster targets. When searching for data for [KIC 5112687](http://simbad.u-strasbg.fr/simbad/sim-basic?Ident=KIC+5112687), I noticed that the light curve and TPF data for [KIC 5112705](http://simbad.u-strasbg.fr/simbad/sim-id?Ident=KIC+5112705) were also returned. These targets are close, but visibly identifiable as being a little over 4" apart in Simbad (as linked), unlike the example above.

More worryingly,

```python
search_lightcurve('KIC 5112687', mission='Kepler', cadence='long', quarter=11)
```

and

```python
search_lightcurve('KIC 5112705', mission='Kepler', cadence='long', quarter=11)
```

both return KIC 5112687 as their first result:

```
# | observation | target_name | productFilename | distance
-- | -- | -- | -- | --
0 | Kepler Quarter 11 | kplr005112687 | kplr005112687-2012004120508_llc.fits | 0.0
1 | Kepler Quarter 11 | kplr005112705 | kplr005112705-2012004120508_llc.fits | 0.0
```

From my perspective, I think that when searching with KIC/TIC ID, then something like `strict_target_name` should be the default, especially given the fact that it seems as though the search function is currently preferencing one of these targets over the other.

**Comment by brackham** [comment]
Just adding another example of this issue that I've encountered.

    sr = lk.search_lightcurvefile('KIC5385723', mission='Kepler')

returns the following SearchResult:

<img width="603" alt="Screen Shot 2020-07-21 at 10 36 28 AM" src="https://user-images.githubusercontent.com/16804223/88068382-1c370480-cb3e-11ea-9f78-2dfa7844c0d3.png">

I'm able to narrow it down to what I want with

    idx_good = np.where(sr.target_name == 'kplr005385723')
    sr = sr[idx_good]

But I agree that it would be nice to strictly search for light curves of a specific target (either by default or via passing an argument).

**Comment by barentsen** [comment]
@astrobel @brackham Thank you for providing these additional examples. They will be very useful to add as unit tests.

I hear you loud and clear! We need to fix this!

**Comment by barentsen** [comment]
Good news: I believe I have implemented a solution to this problem in #796.

**Comment by barentsen** [comment]
I merged #796 -- this issue should now be fixed!

The fix is included in the `v2.0a1` pre-release of Lightkurve v2.0.  You can install it by adding `--pre` to pip, i.e.:
```!pip install lightkurve --upgrade --pre```

(Note: the alpha pre-release of v2.0 may likely break code in other ways.  See `CHANGES.rst` for a summary of major changes.)

**Comment by astrobel** [comment]
Looks like it's working perfectly! Thanks so much for this fix @barentsen!
